tests/test_create_pod_attaching_multiple_pvc_start_IO.py

```diff
@@ -0,0 +1,376 @@
+    open(TEMP_YAML, 'w').close()
+    sleep(5)
+    # Check for Pool creation success from ceph and OC
+    try:
+        CBP.get(ceph_block_pool)
+        log.info(f"Ceph Block Pool {ceph_block_pool} creation success")
+    except exceptions.CommandFailed:
+        assert log.info(
+            f'Exception BlockPool {ceph_block_pool} does not exists '
+        )
+        return False
+
+
+def create_rbd_secret():
+    """
+    Get rbd secret from ceph and create secret in oc
+
+    """
+    secret_data = {}
+    cmd_out = ocs.ocp.exec_ceph_cmd(ceph_cmd="ceph auth get-key client.admin")
+    key = cmd_out.get('key')
+    admin_key_byte = base64.b64encode(key.encode("utf-8"))
+    secret_data['base64_encoded_admin_password'] = str(admin_key_byte, "utf-8")
+    log.info(f"secret_data =  {secret_data}")
+
+    template = os.path.join(TEMPLATES_DIR, "secret.yaml")
+    log.info(f"Create secret for RBD storageclass")
+    data = templating.generate_yaml_from_jinja2_template_with_data(
+        template, **secret_data
+    )
+    secret_name = data['metadata']['name']
+    with open(TEMP_YAML, 'w') as yaml_file:
+        yaml.dump(data, yaml_file)
+    assert SEC.create(yaml_file=TEMP_YAML)
+    open(TEMP_YAML, 'w').close()
+    sleep(5)
+    try:
+        # Check for secret creation success from oc
+        SEC.get(secret_name)
+        log.info(f"Secret {secret_name} creation success")
+        return True
+    except exceptions.CommandFailed:
+        assert log.info(f'Exception Secret {secret_name} does not exists ')
+        return False
+
+
+def create_rbd_storage_class():
+    """
+    Create Storage class in oc
+
+    """
+    template = os.path.join(TEMPLATES_DIR, "storageclass.yaml")
+    log.info(f"Create rbd csi storageclass")
+    data = templating.generate_yaml_from_jinja2_template_with_data(
+        template
+    )
+    storage_class_name = data['metadata']['name']
+    with open(TEMP_YAML, 'w') as yaml_file:
+        yaml.dump(data, yaml_file)
+    assert SC.create(yaml_file=TEMP_YAML)
+    open(TEMP_YAML, 'w').close()
+    sleep(5)
+    try:
+        # Check rbd storageclass creation success from OC
+        SC.get(storage_class_name)
+        log.info(f"SC {storage_class_name} creation success")
+        return True
+    except exceptions.CommandFailed:
+        log.info(f'Exception SC {storage_class_name} does not exists ')
+        return False
+
+
+def create_persistent_volume(volName):
+    """
+    Create a new Persistent Volume
+    """
+    rbd_pvc_data = {}
+    rbd_pvc_data['rbd_pvc_name'] = volName
+    log.info(f"rbd_pvc_data {rbd_pvc_data}")
+    template = os.path.join(TEMPLATES_DIR, "pvc.yaml")
+    log.info(f"Create Persistent Volume Claim ")
+    data = templating.generate_yaml_from_jinja2_template_with_data(
+        template, **rbd_pvc_data
+    )
+    with open(TEMP_YAML, 'w') as yaml_file:
+        yaml.dump(data, yaml_file)
+    assert PVC.create(yaml_file=TEMP_YAML)
+    open(TEMP_YAML, 'w').close()
```

Comment:
no need for that.
This basically clear the temporary file content but, if you will open it up again for write 'w' you will override its content

Reply:
Am loading multiple yaml file in temp during the testing (block_pool_creation, secret, storageclass, pvc, cirros), so after every service creation closing the temp which clears and loading with next required yaml.

Reply:
so next time you'll open that file for 'w' it will erase its content - no need for that

Reply:
Thanks Raz, I just got some info about why its best practice to use "with" while opening a file, incase of with the file will be auto closed once the code execution control is outside with loop.